packages/gatsby/src/components/search/Results.js

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import {
+  Pagination,
+  CurrentRefinements,
+  Stats,
+  connectHits,
+  createConnector,
+} from 'react-instantsearch-dom';
+
+import Hit         from './Hit';
+import { isEmpty } from './util';
+
+const body = document.querySelector('body');
```

Comment:
I think the reason why `document` isn't freely available is that Gatsby also uses SSR, and the document doesn't exist there. Cf #211 which moved the `location` access into a `useLayoutEffect` call to only call this code on mount.

Reply:
Yep, thank you for the hint! I have moved the original code, stripped i18n and made the search work somehow first (there is no CSS now, warnings and other issues). I saw that amendments like this will be needed. In Jekyll they bind React elements to various HTML elements explicitly, this is done due to necessity, because of Jekyll design. In Gatsby this is no longer needed I believe, you should insert your components into proper places in React tree instead.